crates/polars-pipe/src/executors/sinks/file_sink.rs

```diff
@@ -117,6 +129,47 @@ impl IpcSink {
     }
 }
 
+#[cfg(feature = "csv")]
+pub struct CsvSink {}
+#[cfg(feature = "csv")]
+impl CsvSink {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(path: &Path, options: CsvWriterOptions, schema: &Schema) -> PolarsResult<FilesSink> {
+        let file = std::fs::File::create(path)?;
+        let writer = CsvWriter::new(file)
+            .has_header(options.has_header)
+            .with_delimiter(options.serialize_options.delimiter)
+            .with_line_terminator(options.serialize_options.line_terminator)
+            .with_quoting_char(options.serialize_options.quote)
+            .with_batch_size(options.batch_size)
+            .with_datetime_format(options.serialize_options.datetime_format)
+            .with_date_format(options.serialize_options.date_format)
+            .with_time_format(options.serialize_options.time_format)
+            .with_float_precision(options.serialize_options.float_precision)
+            .with_null_value(options.serialize_options.null)
+            .with_quote_style(options.serialize_options.quote_style)
+            .batched(schema)?;
+
+        let writer = Box::new(writer) as Box<dyn SinkWriter + Send + Sync>;
+
+        let morsels_per_sink = morsels_per_sink();
+        let backpressure = morsels_per_sink * 2;
+        let (sender, receiver) = bounded(backpressure);
+
+        let io_thread_handle = Arc::new(Some(init_writer_thread(
+            receiver,
+            writer,
+            options.maintain_order,
+            morsels_per_sink,
+        )));
+
+        Ok(FilesSink {
+            sender,
+            io_thread_handle,
+        })
+    }
+}
+
 #[cfg(any(feature = "parquet", feature = "ipc"))]
```

Comment:
`[cfg(any(feature = "parquet", feature = "ipc", feature="csv"))]`
This method is also required for csv feature